### ISSUE TITLE
In Electron version, try to read backend databases directly using Node.js streams

### ISSUE THREAD

**Issue by tjayrush** [TrueBlocks/trueblocks-explorer#93]
This function ( https://github.com/TrueBlocks/trueblocks-explorer/blob/master/src/ui/modules/core/index.ts#L65) centralizes access to the backend. We did this in anticipation of 'reading the backend files directly into memory' under the Electron application using ideas (potentially) as described here: https://nodejs.dev/learn/nodejs-streams.

I'd like to start with the 'names' database which a simple binary file generated by the backend from an installed text file on first run after installation. The format of the binary file is fairly simple according to this code in the backend: https://github.com/TrueBlocks/trueblocks-core/blob/master/src/libs/utillib/accountname.cpp#L313

Basically native types (ints, bools, etc) are stored directly. Strings are stored with leading uint length followed by the bytes of the string. Types are described here: https://github.com/TrueBlocks/trueblocks-core/blob/master/src/libs/utillib/accountname.h#L30

**Comment by dszlachta** [comment]
A function to read a local file would have to be written in Electron part of the code and then exposed to UI (frontend). 
It would use Streams, just as you have mentioned. 

UI could read the file as blob (binary), but we would need to write a parser. My only concern is making sure that it uses the same schema as the generator in the backend. In the future you could have it as a separate C++ library, used directly in backend and through WebAssembly in frontend, but I guess it is out of scope now. So we will need good unit tests for the parser. 

Hope this helps.
